moped-editor/src/views/projects/projectView/ProjectComponents/TheMap.js

```diff
@@ -179,12 +157,15 @@ export default function TheMap({
 
     // if multiple features are clicked, we ignore all but one
     const clickedFeature = e.features[0];
+    const clickedFeatureFromAgolGeojson =
+      findFeatureInAgolGeojsonFeatures(clickedFeature);
 
     const newFeature = {
-      geometry: clickedFeature.geometry,
+      geometry: clickedFeatureFromAgolGeojson.geometry,
       properties: {
-        ...clickedFeature.properties,
-        id: clickedFeature.id,
+        ...clickedFeatureFromAgolGeojson.properties,
+        id: clickedFeatureFromAgolGeojson.id,
+        // AGOL data doesn't include layer so we grab it from the clicked Mapbox feature
         _layerId: clickedFeature.layer.id,
```

Comment:
Nice! it will be interesting to see how these meta properties factor into the task of committing all this to the DB 🍿